flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetWindowAggregate.scala

```diff
@@ -281,9 +279,36 @@ class DataSetWindowAggregate(
           .returns(rowTypeInfo)
           .name(aggregateOperatorName)
           .asInstanceOf[DataSet[Any]]
+
+      } else {
+        // non-grouping window
+        val preMapPartitionFunction = createDataSetWindowAggregationMapPartitionFunction(
+          window,
+          namedAggregates,
+          inputType)
+
+        val mapPartitionFunction = createDataSetWindowAggregationMapPartitionFunction(
+          window,
+          namedAggregates,
+          inputType,
+          rowRelDataType,
+          namedProperties,
+          isPreMapPartition = false,
+          isInputCombined = true)
+
+        mappedInput.sortPartition(rowTimeFieldPos, Order.ASCENDING)
+          .mapPartition(preMapPartitionFunction)
+          .sortPartition(windowStartPos, Order.ASCENDING).setParallelism(1)
+          .sortPartition(windowEndPos, Order.ASCENDING).setParallelism(1)
+          .mapPartition(mapPartitionFunction).setParallelism(1)
```

Comment:
I think we can also use `.reduceGroup()` and a `GroupReduceFunction` here. Without `groupBy`, the `GroupReduceFunction` will be executed with parallelism 1.